src/finalisers/system.ts

```diff
@@ -0,0 +1,109 @@
+import Chunk from '../Chunk';
+import { Bundle as MagicStringBundle } from 'magic-string';
+import { OutputOptions } from '../rollup/index';
+
+export default function system (
+	chunk: Chunk,
+	magicString: MagicStringBundle,
+	{ getPath, indentString: t, intro, outro }: {
+		indentString: string;
+		getPath: (name: string) => string;
+		intro: string;
+		outro: string
+	},
+	_options: OutputOptions
+) {
+	const { dependencies, exports } = chunk.getModuleDeclarations();
+
+	const deps = dependencies.map(m => `'${getPath(m.id)}'`);
+
+	const importBindings: string[] = [];
+	let starExcludes: string[];
+	const setters: string[] = [];
+	const varOrConst = chunk.graph.varOrConst;
+
+	dependencies.forEach(dep => {
+		let setter: string[] = [];
+		if (dep.imports) {
+			dep.imports.forEach(specifier => {
+				importBindings.push(specifier.local);
+				if (specifier.imported === '*') {
+					setter.push(`${specifier.local} = module;`);
+				} else {
+					setter.push(`${specifier.local} = module.${specifier.imported};`);
+				}
+			});
+		}
+		if (dep.reexports) {
+			// bulk-reexport form
+			if (dep.reexports.length > 1 || dep.reexports.length === 1 && dep.reexports[0].imported === '*') {
+				setter.push(`${varOrConst} _setter = {};`);
+				// star reexports
+				dep.reexports.forEach(specifier => {
+					if (specifier.imported !== '*')
+						return;
+					// need own exports list for deduping in star export case
+					if (!starExcludes) {
+						starExcludes = exports.map(expt => expt.exported);
+						if (starExcludes.indexOf('default') === -1)
+							starExcludes.push('default');
+						// also include reexport names
+						dependencies.forEach(dep => {
+							dep.reexports.forEach(reexport => {
+								if (reexport.imported !== '*' && starExcludes.indexOf(reexport.reexported) === -1)
+									starExcludes.push(reexport.reexported);
+							});
+						});
+					}
+					setter.push(`for (var _$p in module) {`);
+					setter.push(`${t}if (!starExcludes[_$p]) _setter[_$p] = module[_$p];`);
```

Comment:
Also, there is no single test that ever triggers this code. Which is unfortunate as I have the feeling that e.g. `starExcludes` here should rather be `_starExcludes`? Might also be an interesting test case for the other formats.

Reply:
Good catch. I've included a test for this now.